functions/src/services/teamwork.service.ts

```diff
@@ -28,6 +28,43 @@ export class TeamworkService {
       .catch(error => { throw new Error('Unable to authenticate with Teamwork. Please verify your token and try again later.') });
   }
 
+  async getProjects(token: string) {
+    let authData;
+    try {
+      authData = await this.validateToken(token);
```

Comment:
We should not need to make a request to the authenticate endpoint to obtain `authData`.

The endpoint GET `connections/:connectionId/projects` will have the `uid` (from authentication middleware) and the `connectionId` from the route params.

Using these two, we can use the firestore service to get the connection details from document at this path `/users/${uid}/connections/${connectionId}`, which will provide the `token` and `authData` used in this method.